be/src/olap/rowset/beta_rowset_writer.cpp

```diff
@@ -515,7 +516,8 @@ RowsetSharedPtr BetaRowsetWriter::build() {
         }
 
         if (_segcompaction_worker.get_file_writer()) {
-            static_cast<void>(_segcompaction_worker.get_file_writer()->close());
+            WARN_IF_ERROR(_segcompaction_worker.get_file_writer()->close(),
```

Comment:
Printing a warning here would not be enough. Since data may not be synced to disk if file writers close fail, causing data corruption in silence. I suggest we should return nullptr if shit happens.